ApiDoctor.Validation/Config/DocumentOutlineFile.cs

```diff
@@ -75,19 +88,185 @@ public DocumentHeader()
+        }
+
+        /// <summary>
+        /// Indicates that a header pattern can be repeated multiple times e.g. in the case of multiple examples
+        /// </summary>
+        [JsonProperty("multiple")]
+        public bool Multiple { get; set; }
+
+        /// <summary>
+        /// Specifies the headers that are allowed under this header.
+        /// </summary>
+        [JsonProperty("headers"), JsonConverter(typeof(DocumentHeaderJsonConverter))]
+        public new List<object> ChildHeaders { get; set; }
+
+        public ExpectedHeader Clone()
+        {
+            var header = this.MemberwiseClone() as ExpectedHeader;
+            var childHeaders = new List<object>();
+            foreach (var childHeader in this.ChildHeaders)
+            {
+                if (childHeader is ExpectedHeader expectedChildHeader)
+                    childHeaders.Add(expectedChildHeader.Clone());
+                else
+                    childHeaders.Add(((ConditionalHeader)childHeader).Clone());
+            }
+            header.ChildHeaders = childHeaders;
+            return header;
+        }
+    }
+
+    public class ConditionalHeader
+    {
+        [JsonProperty("condition")]
+        public string Condition { get; set; }
+
+        [JsonProperty("arguments"), JsonConverter(typeof(DocumentHeaderJsonConverter))]
+        public List<object> Arguments { get; set; }
+
+        public ConditionalOperator? Operator
+        {
+            get
+            {
+                return Enum.TryParse(this.Condition, true, out ConditionalOperator op) ? op : null;
+            }
+        }
+
+        public ConditionalHeader Clone()
+        {
+            var header = (ConditionalHeader)this.MemberwiseClone();
+            var arguments = new List<object>();
+            foreach (var arg in this.Arguments)
+            {
+                if (arg is ExpectedHeader headerArg)
+                    arguments.Add(headerArg.Clone());
+                else
+                    arguments.Add(((ConditionalHeader)arg).Clone());
+            }
+            header.Arguments = arguments;
+            return header;
+        }
+    }
+
+    public enum ConditionalOperator
+    {
+        OR,
+        AND
     }
 
-}
+    public class DocumentHeaderJsonConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            serializer.Serialize(writer, value);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType == JsonToken.StartArray)
+            {
+                var jArray = JArray.Load(reader);
+                var allowedHeaders = new List<object>();
+                foreach (var item in jArray)
+                {
+                    bool isConditionalHeader = item.ToString().TryParseJson(out ConditionalHeader conditionalHeader);
```

Comment:
this is going to be slow:
1. parse the json to an abitrary model.
2. serialize to string.
3. parse again, with the custom type.

Can't we parse the collection directly?